hack/devtools/rp-dev-helper.sh

```diff
@@ -28,12 +51,75 @@ copy_digest_tag() {
 VERSION_CONST_FILE="pkg/util/version/const.go"
 # Function to get image name and tag
 get_digest_tag() {
+    if [[ $# -ne 1 ]]; then        
+        echo "Error: $0 <IMAGE_NAME>. Please try again"
+        exit 1
+    fi
     local IMAGE_NAME=$1
     local IMAGE_TAG=$(extract_image_tag "$IMAGE_NAME" "$VERSION_CONST_FILE")
-    echo "$IMAGE_NAME and Tag: $IMAGE_TAG"
-    echo "$IMAGE_TAG"
+    echo $IMAGE_TAG
+}
+
+
+# Function to cleanup all the created resources from the full aro RP dev
+clean_rp_dev_env() {
+    echo "########## Deleting Dev Env in $LOCATION ##########"
+
+    # Validate number of arguments
+    if [ $# -gt 2 ]; then
+        echo "Usage: $0 [LIST_RESOURCE_GROUPS] [LIST_KEYVAULTS]. Please try again"
+        exit 1
+    fi
+    if [[ $# -ne 2 ]]; then
+        echo "Info: Two input arguments were required. Checking two env vars for default values"
+        # Check if AZURE_PREFIX environment variable are set
+        if [[ -z "${AZURE_PREFIX}" ]]; then
+            echo "Error: AZURE_PREFIX environment variable is not set."
+            exit 1
+        fi
+        echo "Info: AZURE_PREFIX=${AZURE_PREFIX}"
+        # Check if LOCATION environment variable are set
+        if [[ -z "$LOCATION" ]]; then
+            echo "Error: LOCATION environment variable is not set."
+            exit 1
+        fi
+        echo "Info: LOCATION=$LOCATION"
+    fi
+    
+    # Convert input strings to arrays
+    eval "rgs=($1)"
+    eval "kvs=($2)"
+
+   if [[ ${#rgs[@]} -eq 0 ]]; then
+        rg_suffixes=("global" "subscription" "gwy-$LOCATION" "aro-$LOCATION")
+        rgs=()
+        for suffix in "${rg_suffixes[@]}"; do
+            rgs+=("${AZURE_PREFIX}-$suffix")
+        done
+        echo "No resource groups were provided. Use default values for list: ${rgs[*]}"
+    fi
+
+     for rg in "${rgs[@]}"; do
+        echo "########## Delete Resource Group $rg in $LOCATION ##########"
+        az group delete --resource-group "$rg" -y
```

Comment:
Looking at exiting functions I see that [`clean_env`](https://github.com/Azure/ARO-RP/blob/master/hack/devtools/deploy-shared-env.sh#L236-L253) also deletes reseourceGroups but I am not sure when it is usable. In this function after you delete the reseourceGroup there is no DNS network under this reseourceGroup and also the PARENT_DOMAIN_RESOURCEGROUP (dns) doesn't have record sets (and NS records) with reseourceGroup name 😵

Reply:
I am personally afraid of the use of `clean_env` in general, because if you have the environment variables set wrong (according to . ./env), it will take out a shared dev environment. We should probably remove clean_env entirely, to be honest. For cleanup of the INT-like RP environments, we can probably just advise to remove the resource groups manually as to avoid accidental scripted deletions of important resources.

Reply:
Would that be enough? ResourceGroups and KeyVaults? 
The ResourceGroups seems like the "parent" resource and when deleting it all the "child" resources are deleted as well. Is that right? I don't want to have any Azure orphan resources :(